### PR TITLE
pylint: uprev pylint and address new consider-using-with warnings

### PR DESCRIPTION
**Pull request by cjp256**
Signed-off-by: Chris Patterson <chris.patterson@canonical.com>

- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

**Comment by cjp256** [comment]
Depends on #40 for other req updates.

**Review by cmatsuoka — APPROVED** [review]

File: craft_providers/multipass/multipass.py [hunk 1]
```diff
@@ -342,33 +342,34 @@ def transfer_source_io(
...
+                    break
+
+                proc.stdin.write(data)
+
+            # Wait until process is complete.
+            try:
+                _, stderr = proc.communicate(timeout=30)
+            except subprocess.TimeoutExpired:
+                proc.kill()
+                _, stderr = proc.communicate()
```
Comment (facundobatista) [inline craft_providers/multipass/multipass.py:364]: Do we still need this second `communicate`, taking in consideration that all `fd`s will be closed by the context manager?

File: craft_providers/multipass/multipass.py [hunk 2]
```diff
@@ -342,33 +342,34 @@ def transfer_source_io(
...
+
+            if proc.returncode != 0:
```
Comment (facundobatista) [inline craft_providers/multipass/multipass.py:366]: Considering that the have the context manager now to enforce that the process is waited and all crisp and shiny, this block about checking the `returncode` probably should NOT be indented inside the context manager? (just leave it as before, I mean)

**Review by facundobatista — APPROVED** [review]
Approved, but please take a look to the questions inline, thanks!

File: craft_providers/multipass/multipass.py (hunk 1 above)
Comment (cjp256) [reply craft_providers/multipass/multipass.py]: Done.

File: craft_providers/multipass/multipass.py (hunk 2 above)
Comment (cjp256) [reply craft_providers/multipass/multipass.py]: Good point, fixed!

**Pull request merged**